lib/src/ruby/module.cc

```diff
@@ -908,13 +917,16 @@ namespace facter { namespace ruby {
         return it->second;
     }
 
-    VALUE module::execute_command(std::string const& command, VALUE failure_default, bool raise, uint32_t timeout)
+    VALUE module::execute_command(std::string const& command, VALUE failure_default, bool raise, uint32_t timeout, bool expand)
     {
         auto const& ruby = api::instance();
-
-        // Expand the command
-        auto expanded = expand_command(command);
-
+        std::string expanded;
+        try {
+        // Expand the command only if expand is true,
+        expanded = expand_command(command, leatherman::util::environment::search_paths(), expand);
+        } catch (const std::exception& e ) {
```

Comment:
What's the exception we can get here? Aren't exceptions treated already in expand_command? Also we shouldn't use `std::cout` to log them.